server/service/src/invoice/inbound_shipment/update/mod.rs

```diff
@@ -77,6 +78,19 @@ pub fn update_inbound_shipment(
         )
     );
 
+    let status = &patch.status.expect("Could not get invoice status.");
```

Comment:
status is optional so all other updates would fail. (only on server startup we have unwrap and expect but once the server is running we try to avoid it and always handle all errors, actix catches the panic for a request and the server doesn't crash but think its still better to not panic in our code)

What you want is:
```rust
if let Some(status) = patch.status {
 // log status change here
}
```
you might also want to check if the status has changed? i.e. that it differs from the existing invoice.